### ISSUE TITLE
Bad storage key value in Kusama runtime

### ISSUE THREAD

**Issue by xlc** [polkadot-fellows/runtimes#104]
There are number of undecoable key value pairs in Kusama. Here are some from Balances.locks

- Key `0xc2261276cc9d1f8598ea4b6a74b15c2f218f26c73add634897550b4003b26bc625dd314d5cf9255ed0f3bdf2589adc292f47aed28c0690f788ed2dbceb1fc3ff`
- Value `0x087374616b696e672000000000000000000000000000000000ffffffff1f64656d6f63726163ffffffffffffffffffffffffffffffffc0b0070002`

- Key `0xc2261276cc9d1f8598ea4b6a74b15c2f218f26c73add634897550b4003b26bc6386a9e3f0e74067eef90a9a281c234e05d6fa9c4a25b20fd03913b81d7fccf13`
- Value `0x047374616b696e672000000000000000000000000000000000ffffffff1f`

- Key `0xc2261276cc9d1f8598ea4b6a74b15c2f218f26c73add634897550b4003b26bc6402fc69a7eb122806b64c9e27735e45009aa922e1ba4675758c75607a96a2172`
- Value `0x047374616b696e672000000000000000000000000000000000ffffffff1f`

- Key `0xc2261276cc9d1f8598ea4b6a74b15c2f218f26c73add634897550b4003b26bc645c200adc666cca26df86e2e4a5aa1af186606b9e2a5b86c14cc954228a3cd00`
- Value `0x047374616b696e672000000000000000000000000000000000ffffffff1f`

- Key `0xc2261276cc9d1f8598ea4b6a74b15c2f218f26c73add634897550b4003b26bc651e634907ab2f0b63eadc4091e366f51bcf6f73763f216dcf716cfe7757813e7`
- Value `0x0464656d6f63726163ffffffffffffffffffffffffffffffffc0b0070002`

- Key `0xc2261276cc9d1f8598ea4b6a74b15c2f218f26c73add634897550b4003b26bc65abb1b03f39287c41885312b50a18dc337de40d15f99c182eba0e5aa6629b9bc`
- Value `0x0464656d6f63726163ffffffffffffffffffffffffffffffff7218030002`

all of the above keys are touched by this block https://kusama.subscan.io/block/20466150?tab=event

**Comment by liamaharon** [comment]
I can pick this one up.

**Comment by liamaharon** [comment]
Created https://github.com/liamaharon/which-block to help us debug these issues.

Finds the block a given storage key was set.

**Comment by liamaharon** [comment]
### Investigation results

All of these are Locks in [an old format from 2019](https://github.com/paritytech/substrate/blob/20a9b15cdbed4bf962a4447e8bfb812f766f2fbc/frame/balances/src/lib.rs#L324-L330) which were not migrated.

The locked amounts are `32` for the first four locks, and `340282366920938463463374607431768211299` for the final two locks.

This old code did not use `Blake2_128Concat` for the keys, so it would be non-trivial to find which accounts these locks are for.

I consulted @bkchr about this, and we decided because 
- the locked amounts are either negligible or non-sensical
- these locks are from 2019 and no one has complained
- only 6 accs are affected

that it is not worth looking into what accounts these locks were for at this point, and we can simply delete the storage.

In the very unlikely scenario these locks are breaking an existing account, the owner can come forward and we can fix their accounts on a case-by-case basis.

I will open a PR shortly removing this storage.

**Comment by xlc** [comment]
With your script, it should be possible to find the blocks that wrote those storages and therefore identify impacted accounts?

**Comment by liamaharon** [comment]
@xlc I've done that. It's a PITA to walk through the traces and figure out what's going on an which accounts are impacted. 

E.g. key `0xc2261276cc9d1f8598ea4b6a74b15c2f218f26c73add634897550b4003b26bc625dd314d5cf9255ed0f3bdf2589adc292f47aed28c0690f788ed2dbceb1fc3ff` was introduced in [block 504000](https://kusama.subscan.io/block/504000).

There're no `Lock` events emitted in this block, or extrinsics containing an obvious account. We could sift through the [State Traces](https://kusama.subscan.io/block/504000?tab=state_tracing) to try to figure out which account the lock is for, but not worth the time doing this for all the locks.

**Comment by xlc** [comment]
504000 enacted referendum 11 so I guess it is related democracy pallet to set lock for aye voters. I have only reported 6 bad entries but I am sure there are lot more impacted accounts. It maybe fine to ignore 6 bad entries but less so when the number of impacted accounts is unknown.

Or if you need an impacted user to request for additional investigation, well, here I am.
`HSNBs8VHxcZiqz9NfSQq2YaznTa8BzSvuEWVe4uTihcGiQN` is a multisig controlled by Acala Foundation, and it have `3.8333` KSM reserved. I am unable to identify the source of the reserves and how to get it unreserved. While it is most likely unrelated to the locks entries, there is good chance the root cause of the unidentified reserves is related to the same bad migration of the democracy pallet.

**Comment by liamaharon** [comment]
> I am sure there are lot more impacted accounts

I checked and indeed, there are more accounts. I thought you had listed all the keys in this issue. I convered my PR into a draft and will take a closer look at how wide the impact is.

> Or if you need an impacted user to request for additional investigation, well, here I am.

Reserves are unrelated to this, but yes the `Currency` reserve system really sucks because of what you described. `fungibles` will improve the state of this, as every hold and freeze comes with a reason.
